tests/testGroupScoreAndSync.js

```diff
@@ -0,0 +1,355 @@
+                callback(null);
+            });
+        });
+        
+    
+        
+    });
+    describe('2. GET to instructor assessments URL group tab for the first assessment', function() {
+        it('should load successfully', function(callback) {
+            request(locals.instructorAssessmentsUrlGroupTab, function(error, response, body) {
+                if (error) {
+                    return callback(error);
+                }
+                if (response.statusCode != 200) {
+                    return callback(new Error('bad status: ' + response.statusCode));
+                }
+                //res = response;
+                page = body;
+                callback(null);
+            });
+        });
+        it('should parse', function() {
+            locals.$ = cheerio.load(page);
+        });
+        it('should have a CSRF token', function() {
+            elemList = locals.$('form input[name="__csrf_token"]');
+            assert.lengthOf(elemList, 6);
+            assert.nestedProperty(elemList[0], 'attribs.value');
+            locals.__csrf_token = elemList[0].attribs.value;
+            assert.isString(locals.__csrf_token);
+        });
+    }); 
+    describe('3. user and group initialization', function(){
+        it('create 3 users',  function(callback) {
+            sqldb.query(sql.generate_and_enroll_3_users, [], function(err, result) {
+                if (ERR(err, callback)) return;
+                
+                assert.lengthOf(result.rows, 3);
+                locals.studentUsers = result.rows.slice(0, 3);
+                locals.groupCreator = locals.studentUsers[0];
+                assert.lengthOf(locals.studentUsers, 3);
+                callback(null);
+            });
+        });
+        it('put 3 users in a group',  function(callback) {
+            var form = {
+                __action: 'addGroup',
+                __csrf_token: locals.__csrf_token,
+                groupname: 'test_group',
+                uids: locals.studentUsers[0].uid+','+ locals.studentUsers[1].uid+',' + locals.studentUsers[2].uid,
+            };
+            request.post({url: locals.instructorAssessmentsUrlGroupTab, form: form, followAllRedirects: true}, function (error, response) {
+                if (error) {
+                    return callback(error);
+                }
+                if (response.statusCode != 200) {
+                    return callback(new Error('bad status: ' + response.statusCode));
+                }
+                callback(null);
+            });    
+        });
+        it('should create the correct group configuration', function(callback) {
+            var params = {
+                assessment_id: locals.assessment_id,
+                groupname: 'test_group',
+            };
+            sqldb.query(sql.select_group_users, params, function(err, result) {
+                if (ERR(err, callback)) return;
+                assert.equal(result.rowCount, 3);
```

Comment:
we should have the callback inside the sqldb query so we don’t finish the test before we assert

Reply:
Fix it.